DIG-996: Automate /etc/hosts step during init-authx

OrdiNeu opened the pull request:
[Jira link](https://candig.atlassian.net/browse/DIG-996)

This automates the inclusion of the `/etc/hosts` file into the `make init-authx` step. Includes the removal of documentation related to the step.

To test:
- [x] Linux build
- [x] Windows build
- [x] MacOS build

OrdiNeu committed:
DIG-996: Automate /etc/hosts step during init-authx

OrdiNeu committed:
DIG-996: Vault startup commands fix for MacOS

OrdiNeu committed:
Fix documentation on hosts editing

shaikh-rashid commented on docs/install-docker.md:
````diff
@@ -242,28 +294,7 @@ make init-docker
 make compose
 ```
 
-### Step 5: Update hosts
````
I am not sure what is going on but your doc is out of date. This is adding changes that were removed from previous PR

shaikh-rashid commented on docs/install-docker.md:
```diff
@@ -136,13 +128,73 @@ make docker-pull
 
 If the command still fails, it may be necessary to disable your local firewall, or edit it to allow requests from all ports used in the Docker stack.
 
+## Option 2: Deploy CanDIGv2 using Docker Swarm
```
This section was removed from previous PR.

shaikh-rashid reviewed (changes requested):
Please review git log and review changes. It may be that your doc is adding sections back that was removed from previous PR

OrdiNeu committed:
DIG-996 doc

daisieh reviewed (changes requested):
Even though vault_setup is currently the place where this issue first manifests, it's actually just coincidence that no other container tries to call another container using the docker.localhost alias in an earlier step. It's not specifically vault-related.

This code change would probably be best made as a new target in Makefile that gets called as, perhaps, part of the init-docker target.

OrdiNeu committed:
DIG-996: Move the hosts setup to outside of the vault step, into its own step as part of init-docker

Also skip the step if it is not required, and warn the user if multiple
IP addresses were detected

daisieh commented on setup_hosts.sh:
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+# Early-abort if this step isn't required
+test=$(grep "	docker\.localhost" /etc/hosts)
```
Take out the tab in the grep phrase: it's not always present.

daisieh commented on setup_hosts.sh:
```diff
@@ -0,0 +1,35 @@
+    echo "More than one IP has been detected. Since this script can't automatically determine which one"
+    echo "will make the build work, please remove all but one of the inserted docker.localhost lines in /etc/hosts."
+    echo $IP >.hosts.tmp2
+    awk '{printf "%s\tdocker.localhost\n", $0}' .hosts.tmp2 >> .hosts.tmp
```
I think you're going to need to loop this somehow; otherwise you just get $IP followed by the docker.localhost part.

OrdiNeu replied:
As far as I can tell, `awk` should include it after every line. What version of `awk` do you have installed?

daisieh replied:
awk version 20200816

OrdiNeu committed:
Swap tab with whitespace

OrdiNeu committed:
[Misc] Fix bug where echo was removing newlines

daisieh reviewed (approved):
Looks good! Works perfectly on my system.

CourtneyGosselin commented:
On windows though we also need to change the windows hosts file when using WSL. Maybe we could add a line that tells the user they need to add the IP added to WSL hosts to the windows as well.

shaikh-rashid commented on Makefile:
```diff
@@ -306,7 +313,7 @@ init-conda:
 #<<<
 .PHONY: init-docker
-init-docker: docker-volumes docker-secrets
+init-docker: docker-volumes docker-secrets hosts-file
```
We would not need to modify hosts file for non development environments (i.e. prod deployments). Therefore it would be preferable to have this as an optional step instead of a default action with init-docker

shaikh-rashid reviewed (changes requested):
Minor change requested

OrdiNeu committed:
Move init-hosts-file into its own step

shaikh-rashid reviewed (approved):
LGTM
